src/AcceptanceTests/TransportEncoding/When_receiving_a_message_with_unknown_transport_encoding.cs

```diff
@@ -26,21 +27,25 @@ public class When_receiving_a_message_with_unknown_transport_encoding : NService
 
                         return bus.Send(new MyMessage { Id = ctx.OriginalMessageId }, sendOptions);
                     }))
-                    .WithEndpoint<Receiver>()
-                    .Done(ctx =>
+                    .WithEndpoint<Receiver>(b => b.When((bus, ctx) =>
                     {
-                        var namespaceManager = new NamespaceManagerAdapter(NamespaceManager.CreateFromConnectionString(Environment.GetEnvironmentVariable("AzureServiceBusTransport.ConnectionString"))); // get connection string from test env
-                        var factory = MessagingFactory.CreateAsync(namespaceManager.Address, namespaceManager.Settings.TokenProvider).GetAwaiter().GetResult();
-                        var queueClient = factory.CreateQueueClient("receivingamessagewithunknowntransportencoding.receiver/$DeadLetterQueue", ReceiveMode.ReceiveAndDelete); // queue name from the test name
-                        var message = queueClient.ReceiveAsync().GetAwaiter().GetResult();
-                        var receivedMessageIdMatchesTheOriginal = message.Properties["NServiceBus.MessageId"].ToString() == ctx.OriginalMessageId;
-                        var testRunIdMatchesTheCurrectTestRun = message.Properties["$AcceptanceTesting.TestRunId"] as string == ctx.TestRunId.ToString();
-                        var deliveredOnceOnly = message.DeliveryCount == 1;
-                        ctx.MessageWasMovedToDlq = receivedMessageIdMatchesTheOriginal && testRunIdMatchesTheCurrectTestRun && deliveredOnceOnly;
-
-                        return ctx.MessageWasMovedToDlq;
-                    })
-                    .Run(new RunSettings() {TestExecutionTimeout = TimeSpan.FromMinutes(2)});
+                        // can't apply a message spy pattern here since message will be always poisonous
+                        return Task.Run(async () =>
```

Comment:
It is basically an ErrorSpy on steroids. For some messages they need to go raw receive. Spinning up a raw receiver and still have access to the context. The hack @SeanFeldman did now is starting that raw receive in the When. But I'd say it would be cleaner to do something like

```
var context = new ScenarioContext();
var rawReceive = Task.Run(async => { // rawReceiveWithAccessToContext ));
var scenarioTask = Scenario.Define(context)...
await Task.WhenAll();
```